src/utils/KmlUtils.js

```diff
@@ -0,0 +1,41 @@
+
+import Location from '../models/Location';
+
+const locationsFromPlacemark = (placemark: string): Array<Location> => {
+	var regex = /-?[\d|.|e|E|\+]+,-?[\d|.|e|E|\+]+,-?[\d|.|e|E|\+]+/g;
+	var start = 0;
+	for (let i = 0; i < placemark.length; i++) {
+		var jspace = placemark.indexOf(" ", i);
+		var substr = placemark.substring(i, jspace);
+		if (substr.match(regex) != null) {
+			start = i;
+			break;
+		}
+	}
+
+	var end = 0;
+	for (let i = start; i < placemark.length; i++) {
+		var jspace = placemark.indexOf(" ", i);
+		var substr = placemark.substring(i, jspace);
+		if (substr.match(regex) == null) {
+			end = i;
+			break;
+		}
+		i += substr.length;
+	}
+
+	var coordinateStr = placemark.substring(start, end);
+	var locations = coordinateStr

```

Comment:
Add light comments here to indicate what you're doing in this series of maps / splits / slices, so that it's readable if we ever need to revisit this code.  I understand the data-processing you're doing, but it takes me a few seconds to wrap my head around the nested maps and such.

Also, if you're performing multi-line, nested functionality in a `map` function, it makes it easier to read if you do the following:

````
// previous data processing
.trim() 
.split(" ")
.map(a: DECLARE_A_TYPE => {
  return a.split(",").map(b: DECLARE_A_TYPE => Number(b)).slice(0, 2);
})
// etc., you get the point
````